Toggle added to enable Hosting Locally the Google Fonts

mskapusuz opened the pull request:
### Summary
<!-- Please describe the changes you made. -->
A toggle was added to enable Google Fonts to host locally.

If the toggle is enabled; the Google Fonts should be hosted locally. Before that, we were offering a snippet to enable it. https://docs.themeisle.com/article/1349-how-to-load-neve-fonts-locally#code-snippet

From now on we have toggle + snippet (snippet overrides the toggle)
![Screen Shot 2022-11-20 at 20 35 56](https://user-images.githubusercontent.com/25361626/202917109-4800c089-1d19-44b4-b7d1-66607d0843aa.png)


### Will affect visual aspect of the product
<!-- It includes visual changes? -->
YES

### Screenshots <!-- if applicable -->

### Test instructions
<!-- Describe how this pull request can be tested. -->
- The toggle was added to Customizer -> Typography
- If the toggle is activated; Google fonts should be loaded locally.


Btw: Also, a quick look at the customizer would be good against regression on the customizer overall.

<!-- Issues that this pull request closes. -->
Closes #3671.
<!-- Should look like this: `Closes #1, #2, #3.` . -->

mskapusuz commented on assets/apps/customizer-controls/src/builder/scss/_typography-extra-section.scss:
```diff
@@ -0,0 +1,19 @@
+.control-section-typography_extra_section {
+  display: list-item !important;
+
+  .neve-white-background-control {
+    background: none !important;
+    padding: 15px 30px;
+    position: relative;
+  }
+
+  label {
+    color: rgb(80, 87, 94);
+  }
+
+  .customize-control-description {
+    position: absolute;
+    right: 30px;
+    top: 6px;
+  }
+}
```
Hey @abaicus can you look here closer against is this place ok or is it like a hack? Thank you.

abaicus replied:
@mskapusuz a good solution would be to extend the section constructor and overwrite the `isContextuallyActive` & `attachEvents` methods. This way, the section won't be affected by any unexpected events or callbacks the customizer might fire.

```js
wp.customize.sectionConstructor['typography_extra_section'] = wp.customize.Section.extend({
	// No events for this type of section.
	attachEvents: () => null,
	// Always make the section active.
	isContextuallyActive: () => true,
});
```

mskapusuz replied:
Great. Thank you @abaicus for your review!

abaicus reviewed (approved):
Except for what I mentioned, things look good 👍🏻

irinelenache commented:
@mskapusuz Tested the enhancement and it works fine 🚀 Here are some things regarding compatibilities:

- [ ] If Neve pro is activated, we have [this toggle](https://vertis.d.pr/i/JpMKvp) inside Neve options, that does the same thing. I think the customizer toggle should be synced with this one.
- [ ] Also, the Neve pro option is also synced with [this Otter pro option](https://vertis.d.pr/i/NQOqTt)

selul commented:
@mskapusuz i think the customizer option can share the same key with pro to avoid any complex sync

mskapusuz commented:
@irinelenache Nice catch! 🚀 

I've refactored that to use the option key of the Neve Pro.

Also, I fixed another issue (I discovered a priority issue when Neve Pro is activated):

- If pro activated: before: https://vertis.d.pr/i/Ium2Cl  after the fix: https://vertis.d.pr/i/bU8CnU
- If pro is not activated:  https://vertis.d.pr/i/YqPB4o (no change)

irinelenache commented:
@mskapusuz Checked again and now all the toggle are synced, thank you 🚀
